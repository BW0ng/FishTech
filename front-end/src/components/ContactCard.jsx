
import { Accordion } from "react-bootstrap";
import AdminContact from './AdminContact';
import RegistrantContact from './RegistrantContact';
import TechContact from './TechContact';

function ContactCard(props) {
    return (
        <Accordion.Item eventKey="1">
            <Accordion.Header>Contact Information</Accordion.Header>
            <Accordion.Body>
                <Accordion>
                    <AdminContact data={props.data} />
                    <TechContact data={props.data} />
                    <RegistrantContact data={props.data} />
                </Accordion>
            </Accordion.Body>
        </Accordion.Item>
    );
}

export default ContactCard;