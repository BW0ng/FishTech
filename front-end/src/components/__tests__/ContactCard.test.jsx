import React from 'react';
import renderer from 'react-test-renderer';
import ContactCard from '../ContactCard.jsx';

describe('ContactCard', () => {
    it('renders correctly with full data output', () => {
        let input = { 
            administrativeContact : {
                organization : 'Organization',
                state : 'State',
                country : 'Country'
            },
            technicalContact : {
                organization : 'Organization',
                state : 'State',
                country : 'Country'
            },
            registrant : {
                organization : 'Organization',
                state : 'State',
                country : 'Country'
            },
         };
        const component = renderer.create(<ContactCard data={input} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
        // On the first run of this test, Jest will generate a snapshot file automatically.
      });

      it('renders correctly with missing the administrativeContact', () => {
        let input = { 
            technicalContact : {
                organization : 'Organization',
                state : 'State',
                country : 'Country'
            },
            registrant : {
                organization : 'Organization',
                state : 'State',
                country : 'Country'
            },
         };
        const component = renderer.create(<ContactCard data={input} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
        // On the first run of this test, Jest will generate a snapshot file automatically.
      });

      it('renders correctly with missing the technicalContact', () => {
        let input = { 
            administrativeContact : {
                organization : 'Organization',
                state : 'State',
                country : 'Country'
            },
            registrant : {
                organization : 'Organization',
                state : 'State',
                country : 'Country'
            },
         };
        const component = renderer.create(<ContactCard data={input} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
        // On the first run of this test, Jest will generate a snapshot file automatically.
      });

      it('renders correctly with missing the registrant', () => {
        let input = { 
            administrativeContact : {
                organization : 'Organization',
                state : 'State',
                country : 'Country'
            },
            technicalContact : {
                organization : 'Organization',
                state : 'State',
                country : 'Country'
            },
         };
        const component = renderer.create(<ContactCard data={input} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
        // On the first run of this test, Jest will generate a snapshot file automatically.
      });
});