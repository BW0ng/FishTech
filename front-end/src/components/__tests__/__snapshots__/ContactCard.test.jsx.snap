// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactCard renders correctly with full data output 1`] = `
<div
  className="accordion-item"
>
  <h2
    className="accordion-header"
  >
    <button
      aria-expanded={false}
      className="accordion-button collapsed"
      onClick={[Function]}
      type="button"
    >
      Contact Information
    </button>
  </h2>
  <div
    aria-expanded={null}
    className="accordion-collapse collapse"
  >
    <div
      className="accordion-body"
    >
      <div
        className="accordion"
      >
        <div
          className="accordion-item"
        >
          <h2
            className="accordion-header"
          >
            <button
              aria-expanded={false}
              className="accordion-button collapsed"
              onClick={[Function]}
              type="button"
            >
              Administration Contact
            </button>
          </h2>
          <div
            aria-expanded={null}
            className="accordion-collapse collapse"
          >
            <div
              className="accordion-body"
            >
              <strong>
                Organization: 
              </strong>
              Organization
              <br />
              <strong>
                State: 
              </strong>
              State
              <br />
              <strong>
                Country: 
              </strong>
              Country
              <br />
            </div>
          </div>
        </div>
        <div
          className="accordion-item"
        >
          <h2
            className="accordion-header"
          >
            <button
              aria-expanded={false}
              className="accordion-button collapsed"
              onClick={[Function]}
              type="button"
            >
              Technical Contact
            </button>
          </h2>
          <div
            aria-expanded={null}
            className="accordion-collapse collapse"
          >
            <div
              className="accordion-body"
            >
              <strong>
                Organization: 
              </strong>
              Organization
              <br />
              <strong>
                State: 
              </strong>
              State
              <br />
              <strong>
                Country: 
              </strong>
              Country
              <br />
            </div>
          </div>
        </div>
        <div
          className="accordion-item"
        >
          <h2
            className="accordion-header"
          >
            <button
              aria-expanded={false}
              className="accordion-button collapsed"
              onClick={[Function]}
              type="button"
            >
              Registrant Contact
            </button>
          </h2>
          <div
            aria-expanded={null}
            className="accordion-collapse collapse"
          >
            <div
              className="accordion-body"
            >
              <strong>
                Organization: 
              </strong>
              Organization
              <br />
              <strong>
                State: 
              </strong>
              State
              <br />
              <strong>
                Country: 
              </strong>
              Country
              <br />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ContactCard renders correctly with missing the administrativeContact 1`] = `
<div
  className="accordion-item"
>
  <h2
    className="accordion-header"
  >
    <button
      aria-expanded={false}
      className="accordion-button collapsed"
      onClick={[Function]}
      type="button"
    >
      Contact Information
    </button>
  </h2>
  <div
    aria-expanded={null}
    className="accordion-collapse collapse"
  >
    <div
      className="accordion-body"
    >
      <div
        className="accordion"
      >
        <div
          className="accordion-item"
        >
          <h2
            className="accordion-header"
          >
            <button
              aria-expanded={false}
              className="accordion-button collapsed"
              onClick={[Function]}
              type="button"
            >
              Technical Contact
            </button>
          </h2>
          <div
            aria-expanded={null}
            className="accordion-collapse collapse"
          >
            <div
              className="accordion-body"
            >
              <strong>
                Organization: 
              </strong>
              Organization
              <br />
              <strong>
                State: 
              </strong>
              State
              <br />
              <strong>
                Country: 
              </strong>
              Country
              <br />
            </div>
          </div>
        </div>
        <div
          className="accordion-item"
        >
          <h2
            className="accordion-header"
          >
            <button
              aria-expanded={false}
              className="accordion-button collapsed"
              onClick={[Function]}
              type="button"
            >
              Registrant Contact
            </button>
          </h2>
          <div
            aria-expanded={null}
            className="accordion-collapse collapse"
          >
            <div
              className="accordion-body"
            >
              <strong>
                Organization: 
              </strong>
              Organization
              <br />
              <strong>
                State: 
              </strong>
              State
              <br />
              <strong>
                Country: 
              </strong>
              Country
              <br />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ContactCard renders correctly with missing the registrant 1`] = `
<div
  className="accordion-item"
>
  <h2
    className="accordion-header"
  >
    <button
      aria-expanded={false}
      className="accordion-button collapsed"
      onClick={[Function]}
      type="button"
    >
      Contact Information
    </button>
  </h2>
  <div
    aria-expanded={null}
    className="accordion-collapse collapse"
  >
    <div
      className="accordion-body"
    >
      <div
        className="accordion"
      >
        <div
          className="accordion-item"
        >
          <h2
            className="accordion-header"
          >
            <button
              aria-expanded={false}
              className="accordion-button collapsed"
              onClick={[Function]}
              type="button"
            >
              Administration Contact
            </button>
          </h2>
          <div
            aria-expanded={null}
            className="accordion-collapse collapse"
          >
            <div
              className="accordion-body"
            >
              <strong>
                Organization: 
              </strong>
              Organization
              <br />
              <strong>
                State: 
              </strong>
              State
              <br />
              <strong>
                Country: 
              </strong>
              Country
              <br />
            </div>
          </div>
        </div>
        <div
          className="accordion-item"
        >
          <h2
            className="accordion-header"
          >
            <button
              aria-expanded={false}
              className="accordion-button collapsed"
              onClick={[Function]}
              type="button"
            >
              Technical Contact
            </button>
          </h2>
          <div
            aria-expanded={null}
            className="accordion-collapse collapse"
          >
            <div
              className="accordion-body"
            >
              <strong>
                Organization: 
              </strong>
              Organization
              <br />
              <strong>
                State: 
              </strong>
              State
              <br />
              <strong>
                Country: 
              </strong>
              Country
              <br />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ContactCard renders correctly with missing the technicalContact 1`] = `
<div
  className="accordion-item"
>
  <h2
    className="accordion-header"
  >
    <button
      aria-expanded={false}
      className="accordion-button collapsed"
      onClick={[Function]}
      type="button"
    >
      Contact Information
    </button>
  </h2>
  <div
    aria-expanded={null}
    className="accordion-collapse collapse"
  >
    <div
      className="accordion-body"
    >
      <div
        className="accordion"
      >
        <div
          className="accordion-item"
        >
          <h2
            className="accordion-header"
          >
            <button
              aria-expanded={false}
              className="accordion-button collapsed"
              onClick={[Function]}
              type="button"
            >
              Administration Contact
            </button>
          </h2>
          <div
            aria-expanded={null}
            className="accordion-collapse collapse"
          >
            <div
              className="accordion-body"
            >
              <strong>
                Organization: 
              </strong>
              Organization
              <br />
              <strong>
                State: 
              </strong>
              State
              <br />
              <strong>
                Country: 
              </strong>
              Country
              <br />
            </div>
          </div>
        </div>
        <div
          className="accordion-item"
        >
          <h2
            className="accordion-header"
          >
            <button
              aria-expanded={false}
              className="accordion-button collapsed"
              onClick={[Function]}
              type="button"
            >
              Registrant Contact
            </button>
          </h2>
          <div
            aria-expanded={null}
            className="accordion-collapse collapse"
          >
            <div
              className="accordion-body"
            >
              <strong>
                Organization: 
              </strong>
              Organization
              <br />
              <strong>
                State: 
              </strong>
              State
              <br />
              <strong>
                Country: 
              </strong>
              Country
              <br />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
